keyboards/handwired/mingwahlu/mingwahlumacro/keymaps/default/keymap.c

```diff
@@ -0,0 +1,184 @@
+    _BASE,
+    _FN
+};
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    QMKBEST = SAFE_RANGE,
+    QMKURL,
+    M_NUTS,
+    M_RESET,
+    M_CLOSETAB,
+    M_REOPENTAB,
+    M_NEWTAB,
+    M_SWITCHTABS,
+    M_ALTTAB,
+    M_ALTF4,
+    M_UMARUBM,
+    M_UMARUEAT,
+    M_UMARUMAD,
+    M_UMARUHAPPY,
+    M_UMARUSAD,
+    M_UMARUGAME,
+    M_UMARUSLEEP
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+        M_UMARUHAPPY, M_UMARUSAD, M_UMARUMAD, MO(_FN),
+        M_UMARUEAT, M_UMARUGAME, M_UMARUBM, M_UMARUSLEEP
+    ),
+    [_FN] = LAYOUT(
+        M_ALTF4, KC_F5, M_NUTS, MO(_FN),
+        M_CLOSETAB, M_REOPENTAB, M_NEWTAB, M_SWITCHTABS
+    )
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case QMKBEST:
+            if (record->event.pressed) {
+                // when keycode QMKBEST is pressed
+                SEND_STRING("QMK is the best thing ever!");
+            } else {
+                // when keycode QMKBEST is released
+            }
+            break;
+        case QMKURL:
+            if (record->event.pressed) {
+                // when keycode QMKURL is pressed
+                SEND_STRING("https://qmk.fm/\n");
+            } else {
+                // when keycode QMKURL is released
+            }
+            break;
+        case M_NUTS:
+            if (record->event.pressed) {
+                SEND_STRING("deeznuts");
+                tap_code(KC_ENTER);
+            } else {
+            }
+            break;
+        case M_RESET:
+            if (record->event.pressed) {
+                reset_keyboard();
+            } else {
+            }
+            break;
+        case M_CLOSETAB:
+            if(record->event.pressed) {
+                SEND_STRING(SS_LCTL("w"));
+            } else {
+
+            }
+            break;
+        case M_REOPENTAB:
+            if(record->event.pressed) {
+                SEND_STRING(SS_LCTL(SS_LSFT("t")));
+            } else {
+
+            }
+            break;
+        case M_NEWTAB:
+            if(record->event.pressed) {
+                SEND_STRING(SS_LCTL("t"));
+            } else {
+
+            }
+            break;
+        case M_SWITCHTABS:
+            if(record->event.pressed) {
+                register_code(KC_LCTL);
+                tap_code(KC_TAB);
+            } else {
+               unregister_code(KC_LCTL);
+            }
+            break;
+        case M_ALTTAB:
+            if(record->event.pressed) {
+                register_code(KC_LALT);
+                tap_code(KC_TAB);
+            } else {
+                unregister_code(KC_LALT);
+            }
+            break;
+        case M_ALTF4:
+            if(record->event.pressed) {
+                register_code(KC_LALT);
+                tap_code(KC_F4);
+            } else {
+               unregister_code(KC_LALT);
+            }
```

Comment:
This can be replaced with `LALT(KC_F4)` in the keymap.

Reply:
macros have been removed, and we are now using LALT(KC_F4) in the keymap.